.claude/hooks/web-research-extended-gate.sh

```
@@ -0,0 +1,69 @@
+
+if command -v jq >/dev/null 2>&1; then
+  FILE_PATH=$(echo "$INPUT" | jq -r '.tool_input.file_path // empty' 2>/dev/null)
+  CONTENT=$(echo "$INPUT" | jq -r '.tool_input.new_string // .tool_input.content // empty' 2>/dev/null)
+else
+  FILE_PATH=""
+  CONTENT=$(echo "$INPUT" | sed -n 's/.*"\(new_string\|content\)"[[:space:]]*:[[:space:]]*"\([^"]*\)".*/\2/p' | head -1)
+fi
+
+# Skip hook infrastructure
+case "$FILE_PATH" in
+  */.claude/hooks/*|.claude/hooks/*|./.claude/hooks/*) exit 0 ;;
```

Shellcheck (0.11.0)

[warning] This pattern always overrides a later one on line 25.

(SC2221)